src/metabase/sync/analyze/classifiers/name.clj

```diff
@@ -114,12 +116,11 @@
 (s/defn ^:private special-type-for-name-and-base-type :- (s/maybe su/FieldType)
   "If `name` and `base-type` matches a known pattern, return the `special_type` we should assign to it."
   [field-name :- su/NonBlankString, base-type :- su/FieldType]
-  (or (when (= "id" (str/lower-case field-name)) :type/PK)
-      (some (fn [[name-pattern valid-base-types special-type]]
-              (when (and (some (partial isa? base-type) valid-base-types)
-                         (re-find name-pattern (str/lower-case field-name)))
-                special-type))
-            pattern+base-types+special-type)))
+  (some (fn [[name-pattern valid-base-types special-type]]
+          (when (and (some (partial isa? base-type) valid-base-types)
+                     (re-find name-pattern (str/lower-case field-name)))
```

Comment:
This will lower-case the field name worst case 70 times for each invocation. Seems like it'd be better to let-bind it or make the regex case in-sensitive, i.e. `(?i)`.

As a separate discussion (not something to be addressed in this PR), I think eventually we'll need to get away from comparing table/column names in a case-insensitive manner (or forcing upper/lower casing). I think in general it's not a very common use case, but databases don't require that same uniqueness and it can lead to issues like https://github.com/metabase/metabase/issues/7923.